test/framework/shoot_utils.go

```diff
@@ -276,6 +278,14 @@ func setShootGeneralSettings(shoot *gardencorev1beta1.Shoot, cfg *ShootCreationC
 	}
 }
 
+// setShootHighAvailabilitySettings sets the Shoot's HighAvailability settings from the given config
+func setShootHighAvailabilitySettings(shoot *gardencorev1beta1.Shoot, cfg *ShootCreationConfig) {
+	if StringSet(cfg.failureToleranceType) {
+		shoot.Spec.ControlPlane.HighAvailability.FailureTolerance.Type = cfg.failureToleranceType
```

Comment:
If I read it correctly, this code would hit a nil pointer. Both, `ControlPlane` and `HightAvailability` can be `nil` and most likely they are not set yet.